v2/src/log_context_property_type_wchar_t_ptr.c

```diff
@@ -0,0 +1,148 @@
+{
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_012: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr. ]*/
+    return LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr;
+}
+
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(void* dst_value, size_t count, const wchar_t* format, ...)
+{
+    int result;
+
+    if (
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_013: [ If dst_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+        (dst_value == NULL) ||
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_014: [ If format is NULL, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+        (format == NULL)
+        )
+    {
+        (void)printf("Invalid arguments: void* dst_value=%p, const wchar_t* format=%ls\r\n",
+            dst_value, MU_WP_OR_NULL(format));
+        result = MU_FAILURE;
+    }
+    else
+    {
+        va_list args;
+        va_start(args, format);
+
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_015: [ LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall initialize by calling vswprintf the memory at dst_value with the wprintf style formatted wchar_t string given by format and the arguments in .... ]*/
+        int vswprintf_result = vswprintf(dst_value, count, format, args);
+
+        if (vswprintf_result < 0)
+        {
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_017: [ If formatting the string fails, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+            result = MU_FAILURE;
+        }
+        else
+        {
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_016: [ LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall succeed and return 0. ]*/
+            result = 0;
+        }
+
+        va_end(args);
+    }
+
+    return result;
+}
+
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(const wchar_t* format, ...)
+{
+    va_list args;
+    va_start(args, format);
+
+    wchar_t buffer[LOG_MAX_WCHAR_STRING_LENGTH];
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_018: [ If vswprintf fails, LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr) shall return a negative value. ]*/
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_019: [ Otherwise, on success, LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr) shall return the amount of memory needed in number of wide-chracters to store the wprintf style formatted wchar_t string given by format and the arguments in .... ]*/
+    int result = vswprintf(buffer, LOG_MAX_WCHAR_STRING_LENGTH, format, args) * sizeof(wchar_t);
+    if (result >= 0)
+    {
+        result += sizeof(wchar_t);
+    }
+    else
+    {
+        (void)printf("wchar_t string length exceeds LOG_MAX_WCHAR_STRING_LENGTH, wchar string has been truncated");
```

Comment:
>printf [](http://example.com/codeflow?start=14&length=6)

should print the number of characters.

Reply:
have deleted this printf, vswprintf doesn't state it can return number of wchars should be written by passing NULL and 0, so I use LOG_MAX_WCHAR_STRING_LENGTH. If that works, just remove this print to keep the same the ascii char.

Reply:
don't actually know the size here, vswprintf return size is limit by the second parameter

Reply:
vswprint returns a negative value in cases when actual length is greater than the length passed or an output error occurs, can't distinguish between them. removed this line, error will occur if result < 0 in log.context.h